### PR TITLE
Remove native_functions.yaml dependency from DistributionNormal.cu

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67875
* __->__ #67874
* #67721
* #66980
* #66979

Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

### CODE REVIEW

File: aten/src/ATen/native/UnaryOps.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 namespace at {
 class Tensor;
```
Comment (Lezcano) [inline aten/src/ATen/native/UnaryOps.h:8]: Nit. I believe this is not needed anymore.

File: aten/src/ATen/native/UnaryOps.h [hunk 2]
```diff
@@ -6,6 +6,7 @@
...
+class TensorBase;
 struct TensorIterator;
```
Comment (Lezcano) [inline aten/src/ATen/native/UnaryOps.h:10]: Same with this line.

File: aten/src/ATen/native/UnaryOps.h (hunk 1 above)
Comment (peterbell10) [reply aten/src/ATen/native/UnaryOps.h:8]: It's used by `multinomial_with_replacement_stub`.

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Review by dagitses — APPROVED** [review]

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by dagitses** [comment]
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

**Pull request closed without merging**